perllib/FixMyStreet/App/Controller/Auth.pm

```diff
@@ -444,6 +446,37 @@ sub change_password : Local {
 
 }
 
+sub get_csrf_token : Private {
+    my ( $self, $c ) = @_;
+
+    my $time = $c->stash->{csrf_time} || time();
+    my $hash = hmac_sha1("$time-" . ($c->sessionid || ""), $c->model('DB::Secret')->get);
+    $hash = encode_base64($hash, "");
+    my $token = "$time-$hash";
+    $c->stash->{csrf_token} = $token unless $c->stash->{csrf_time};
+    return $token;
+}
+
+sub check_csrf_token : Private {
+    my ( $self, $c ) = @_;
+
+    my $token = $c->get_param('token') || "";
+    $token =~ s/ /+/g;
+    my ($time, $hash) = split /-/, $token;
+    $c->stash->{csrf_time} = $time;
+    $c->detach('no_csrf_token')
+        unless $token
+            && $time > time() - 3600
+            && $token eq $c->forward('get_csrf_token');
+    delete $c->stash->{csrf_time};
+}
+
+sub no_csrf_token : Private {
```

Comment:
It's not used for anything once you're here; it's only deleted in the check in case there's a future call to get_csrf_token (which I don't think there is).